game-core/src/main/java/games/strategy/triplea/delegate/battle/MustFightBattle.java

```diff
@@ -1699,6 +1701,25 @@ public void execute(final ExecutionStack stack, final IDelegateBridge bridge) {
           public void execute(final ExecutionStack stack, final IDelegateBridge bridge) {
             if (!isOver) {
               round++;
+              if (round > MAX_INFINITE_ROUNDS) {
+                // the battle appears to be in an infinite loop
+                throw new IllegalStateException(
+                    "Round 10,000 reached in a battle. Something must be wrong."
```

Comment:
What do you think of adding to the message "Please report this to TripleA"?
A line break between this first line and the unit listing could make it an easier message to read as well.